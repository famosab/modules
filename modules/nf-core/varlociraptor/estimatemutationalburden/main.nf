process VARLOCIRAPTOR_ESTIMATEMUTATIONALBURDEN {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/varlociraptor:8.7.3--ha8ac579_2':
        'biocontainers/varlociraptor:8.7.3--ha8ac579_2' }"

    input:
    tuple val(meta), path(vcf)
    val(plot_mode)

    output:
    tuple val(meta), path("*.json"), emit: json
    path "versions.yml"            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    varlociraptor estimate mutational-burden \\
        --plot-mode ${plot_mode} \\
        ${args} \\
        < ${vcf} \\
        > ${prefix}.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        varlociraptor: \$(echo \$(varlociraptor --version 2>&1) | sed 's/^.*varlociraptor //; s/:.*\$//' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        varlociraptor: \$(echo \$(varlociraptor --version 2>&1) | sed 's/^.*varlociraptor //; s/:.*\$//' )
    END_VERSIONS
    """
}
