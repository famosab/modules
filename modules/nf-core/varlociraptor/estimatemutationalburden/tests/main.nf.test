nextflow_process {

    name "Test Process VARLOCIRAPTOR_ESTIMATEMUTATIONALBURDEN"
    script "../main.nf"
    process "VARLOCIRAPTOR_ESTIMATEMUTATIONALBURDEN"

    tag "modules"
    tag "modules_nfcore"
    tag "varlociraptor"
    tag "varlociraptor/estimatemutationalburden"
    tag "varlociraptor/estimatealignmentproperties"
    tag "varlociraptor/preprocess"
    tag "varlociraptor/callvariants"
    tag "bcftools/view"
    tag "snpeff/download"
    tag "snpeff/snpeff"

    setup {
        run("VARLOCIRAPTOR_ESTIMATEALIGNMENTPROPERTIES") {
            script "../../estimatealignmentproperties/main.nf"
            process {
                """
                input[0] = [
                    [id:'test_normal'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists:true),
                    ]
                input[1] = [
                    [id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists:true)
                    ]
                input[2] = [
                    [id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists:true)
                    ]
                """
            }
        }

        run("VARLOCIRAPTOR_PREPROCESS" ) {
            script "../../preprocess/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [id:'test_normal'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gvcf/test.genome.vcf', checkIfExists:true),
                    ]).collect().join(VARLOCIRAPTOR_ESTIMATEALIGNMENTPROPERTIES.out.alignment_properties_json)
                input[1] = [
                    [id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists:true)
                    ]
                input[2] = [
                    [id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists:true)
                    ]
                """
            }
        }

        run("VARLOCIRAPTOR_CALLVARIANTS") {
            script "../../callvariants/main.nf"
            process {
                """
                input[0] = VARLOCIRAPTOR_PREPROCESS.out.bcf.map{meta1,vcf->[meta1,vcf,[]]}
                input[1] = Channel.of(file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/varlociraptor/scenario.yml', checkIfExists:true))
                input[2] = "normal"
                """
            }
        }

        run("BCFTOOLS_VIEW") {
            script "../../../bcftools/view/main.nf"
            process {
                """
                input[0] = VARLOCIRAPTOR_CALLVARIANTS.out.bcf.map{it -> [ it[0], it[1], [] ]}
                input[1] = []
                input[2] = []
                input[3] = []
                """
            }
        }

        run("SNPEFF_DOWNLOAD") {
            script "../../../snpeff/download/main.nf"
            process {
                """
                input[0] = Channel.of([[id:'GRCh38.105'], 'GRCh38.105'])
                """
            }
        }


        run("SNPEFF_SNPEFF") {
            script "../../../snpeff/snpeff/main.nf"
            process {
                """
                input[0] = BCFTOOLS_VIEW.out.vcf
                input[1] = 'GRCh38.105'
                input[2] = SNPEFF_DOWNLOAD.out.cache
                """
            }
        }

    }

    test("human - vcf") {
        when {
            process {
                """
                input[0] = SNPEFF_SNPEFF.out.vcf
                input[1] = "hist"
                """
            }
        }
        then {
            assert process.success
            assertAll(
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("human - vcf - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = SNPEFF_SNPEFF.out.vcf
                input[1] = "hist"
                """
            }
        }
        then {
            assert process.success
            assertAll(
                { assert snapshot(
                    process.out,
                    path(process.out.versions[0]).yaml
                ).match() }
            )
        }
    }



}
