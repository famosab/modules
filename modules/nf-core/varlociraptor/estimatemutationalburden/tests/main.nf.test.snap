{
    "human - vcf - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_normal"
                        },
                        "test_normal.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,28d5042c15063d0d08d25fa7f0d265b3"
                ],
                "json": [
                    [
                        {
                            "id": "test_normal"
                        },
                        "test_normal.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,28d5042c15063d0d08d25fa7f0d265b3"
                ]
            },
            {
                "VARLOCIRAPTOR_ESTIMATEMUTATIONALBURDEN": {
                    "varlociraptor": "8.7.3"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-07-31T14:53:03.872136"
    }
}